Plugin/Http/Context.php

```diff
@@ -156,20 +156,25 @@ private function getSegmentFromCookie() {
         if ($stdClass === null) {
             $this->logger->debug(sprintf(
                 'Cookie %s has no value',
-                SegmentMapping::COOKIE_NAME
+                SegmentMapping::COOKIE_CATEGORY_MAP
             ));
             return '';
         }
         $segmentMap = get_object_vars($stdClass);
 
+        $hashedCategory = crc32($this->categoryString);
+
         //Check if current category is part of segment mapping
-        if (array_key_exists($this->categoryString, $segmentMap) &&
-            property_exists($segmentMap[$this->categoryString] ,'segmentId')) {
-            $segment = $segmentMap[$this->categoryString];
-            if (property_exists($segment, 'variation')) {
-                return $segment->segmentId . '-' . $segment->variation;
+        if (array_key_exists($hashedCategory, $segmentMap) &&
+            is_numeric($segmentMap[$hashedCategory])) {
+
+            $index = $segmentMap[$hashedCategory];
+            $indexedIds = $this->cookieManager->getCookie(SegmentMapping::COOKIE_SEGMENT_MAP);
+            if ($indexedIds === null || $indexedIds === '') {
+                return '';
             }
-            return $segment->segmentId;
+            $indexedIds = json_decode($indexedIds);
+            return $indexedIds[$index];
```

Comment:
"Because PHP's integer type is signed, and many crc32 checksums will result in negative integers, you need to use the "%u" formatter of sprintf() or printf() to get the string representation of the unsigned crc32 checksum."
I'm not sure what would happen if `$segmentMap[-9879827391872]` is called. If this value comes from JS, it might not come signed.